govcd/cse.go

```diff
@@ -0,0 +1,379 @@
+// of the created cluster cannot be calculated or retrieved.
+func (org *Org) CseCreateKubernetesCluster(clusterData CseClusterSettings, timeout time.Duration) (*CseKubernetesCluster, error) {
+	clusterId, err := org.CseCreateKubernetesClusterAsync(clusterData)
+	if err != nil {
+		return nil, err
+	}
+
+	err = waitUntilClusterIsProvisioned(org.client, clusterId, timeout)
+	if err != nil {
+		return &CseKubernetesCluster{ID: clusterId}, err
+	}
+
+	return getCseKubernetesClusterById(org.client, clusterId)
+}
+
+// CseCreateKubernetesClusterAsync creates a Kubernetes cluster with the data given as input (CseClusterSettings), but does not
+// wait for the creation process to finish, so it doesn't monitor for any errors during the process. It returns just the ID of
+// the created cluster. One can manually check the status of the cluster with VCDClient.CseGetKubernetesClusterById and the result of this method.
+func (org *Org) CseCreateKubernetesClusterAsync(clusterSettings CseClusterSettings) (string, error) {
+	if org == nil {
+		return "", fmt.Errorf("CseCreateKubernetesClusterAsync cannot be called on a nil Organization receiver")
+	}
+
+	cseSubcomponents, err := getCseComponentsVersions(clusterSettings.CseVersion)
+	if err != nil {
+		return "", err
+	}
+
+	internalSettings, err := clusterSettings.toCseClusterSettingsInternal(*org)
+	if err != nil {
+		return "", fmt.Errorf("error creating the CSE Kubernetes cluster: %s", err)
+	}
+
+	payload, err := internalSettings.getUnmarshaledRdePayload()
+	if err != nil {
+		return "", err
+	}
+
+	rde, err := createRdeAndGetFromTask(org.client, "vmware", "capvcdCluster", cseSubcomponents.CapvcdRdeTypeVersion,
+		types.DefinedEntity{
+			EntityType: internalSettings.RdeType.ID,
+			Name:       internalSettings.Name,
+			Entity:     payload,
+		}, &TenantContext{
+			OrgId:   org.Org.ID,
+			OrgName: org.Org.Name,
+		})
+	if err != nil {
+		return "", fmt.Errorf("error creating the CSE Kubernetes cluster: %s", err)
+	}
+
+	return rde.DefinedEntity.ID, nil
+}
+
+// CseGetKubernetesClusterById retrieves a CSE Kubernetes cluster from VCD by its unique ID
+func (vcdClient *VCDClient) CseGetKubernetesClusterById(id string) (*CseKubernetesCluster, error) {
+	return getCseKubernetesClusterById(&vcdClient.Client, id)
+}
+
+// CseGetKubernetesClustersByName retrieves all the CSE Kubernetes clusters from VCD with the given name that belong to the receiver Organization.
+// Note: The clusters retrieved won't have a valid ETag to perform operations on them. Use VCDClient.CseGetKubernetesClusterById for that instead.
+func (org *Org) CseGetKubernetesClustersByName(cseVersion semver.Version, name string) ([]*CseKubernetesCluster, error) {
+	cseSubcomponents, err := getCseComponentsVersions(cseVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	rdes, err := getRdesByName(org.client, "vmware", "capvcdCluster", cseSubcomponents.CapvcdRdeTypeVersion, name)
```

Comment:
Just a word of caution. Maybe these should be constants extracted to the top of file.

Reply:
Done